packages/arb-bridge-eth/contracts/challenge/ExecutionChallenge.sol

```diff
@@ -167,19 +205,29 @@ contract ExecutionChallenge is BisectionChallenge {
         uint64 _gas,
         bytes memory _proof
     ) public asserterAction {
-        verifyPreCondition(
+        bytes32 beforeInbox = Value
+            .newTuplePreImage(_beforeInbox, _beforeInboxValueSize)
+            .hash();
+        ChallengeUtils.ExecutionAssertion memory assertion = ChallengeUtils
+            .ExecutionAssertion(
+            1,
             _beforeHash,
-            _beforeInbox,
-            _beforeInboxValueSize,
+            beforeInbox,
             _afterHash,
             _didInboxInsns,
             _gas,
             _firstMessage,
             _lastMessage,
+            _firstMessage == _lastMessage ? 0 : 1,
```

Comment:
Is this safe?  Suppose `_lastMessage == _firstMessage + 2`.  Then this will succeed if the instruction is a send.  But it should fail.

It looks like this might allow someone to make an assertion that claims to produce more messages than it should, and hide that fact by letting this code replace the true difference in counts with a 1.

Reply:
So first message and last message are both hashes. The execution proof already ensures that they either need to be equal or separated by a single message. I just was taking advantage of that here. Do you think it's good enough to just add a comment explaining this logic?

Reply:
A comment is fine.